src/Sentry/SpanId.cs

```diff
@@ -7,58 +8,82 @@ namespace Sentry;
 /// </summary>
 public readonly struct SpanId : IEquatable<SpanId>, IJsonSerializable
 {
-    private readonly string _value;
+    private static readonly char[] HexChars = { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f' };
+    private static readonly RandomValuesFactory Random = new SynchronizedRandomValuesFactory();
 
-    private const string EmptyValue = "0000000000000000";
+    private readonly long _value;
+
+    private long GetValue() => _value;
 
     /// <summary>
     /// An empty Sentry span ID.
     /// </summary>
-    public static readonly SpanId Empty = new(EmptyValue);
+    public static readonly SpanId Empty = new(0);
 
     /// <summary>
     /// Creates a new instance of a Sentry span Id.
     /// </summary>
-    public SpanId(string value) => _value = value;
-
-    // This method is used to return a string with all zeroes in case
-    // the `_value` is equal to null.
-    // It can be equal to null because this is a struct and always has
-    // a parameterless constructor that evaluates to an instance with
-    // all fields initialized to default values.
-    // Effectively, using this method instead of just referencing `_value`
-    // makes the behavior more consistent, for example:
-    // default(SpanId).ToString() -> "0000000000000000"
-    // default(SpanId) == SpanId.Empty -> true
-    private string GetNormalizedValue() => !string.IsNullOrWhiteSpace(_value)
-        ? _value
-        : EmptyValue;
+    // TODO: mark as internal in version 4
+    public SpanId(string value) => long.TryParse(value, NumberStyles.HexNumber, null, out _value);
+
+    /// <summary>
+    /// Creates a new instance of a Sentry span Id.
+    /// </summary>
+    /// <param name="value"></param>
+    public SpanId(long value) => _value = value;
 
     /// <inheritdoc />
-    public bool Equals(SpanId other) => StringComparer.Ordinal.Equals(
-        GetNormalizedValue(),
-        other.GetNormalizedValue());
+    public bool Equals(SpanId other) => GetValue() == other.GetValue();
```

Comment:
`==` is an operator. Since u're on Equals here, stick to it
```suggestion
    public bool Equals(SpanId other) => GetValue().Equals(other.GetValue());
```